api/models/Post.js

```diff
@@ -0,0 +1,49 @@
+const db = require('../dbConfig/init')
+
+class Post {
+    constructor(data){
+        this.id = data.id,
+        this.title = data.title,
+        this.pname  = data.pname,
+        this.body = data.body;
+    };
+
+    static get all(){
+        return new Promise(async (resolve, reject) => {
+            try {
+                const post = await db.query(`SELECT * FROM posts`);
+                const posts = post.rows.map(r => ({ id: r.id, title: r.title, pname: r.pname, body: r.body}))
+                resolve(posts);
+            } catch (err){
+                reject(`Error retrieving posts: ${err}`)
+            };
+        });
+    };
+
```

Comment:
Everything else seems fine, I just wouldnt use that reject message as you are mixing a string with a json err (which is an object) it may not come out nice. I would say use either {err} alone or type your own error message. Sorry if im nitpicking.